tests/Integration/Behaviour/Features/Context/Domain/CountryFeatureContext.php

```diff
@@ -58,4 +75,71 @@ public function assertCountryNotFound(): void
     {
         $this->assertLastErrorIs(CountryNotFoundException::class);
     }
+
+    /**
+     * @When I add new country :countryReference with following properties:
+     *
+     * @param string $countryReference
+     * @param TableNode $table
+     */
+    public function createCountry(string $countryReference, TableNode $table): void
+    {
+        $data = $this->localizeByRows($table);
+
+        try {
+            $countryId = $this->getCommandBus()->handle(new AddCountryCommand(
+                $data['name'],
+                (string) $data['iso_code'],
+                (int) $data['call_prefix'],
+                (int) $data['default_currency'],
+                (int) $data['zone'],
+                (bool) $data['need_zip_code'],
+                $data['zip_code_format'],
+                (string) $data['address_format'],
+                (bool) $data['is_enabled'],
+                (bool) $data['contains_states'],
+                (bool) $data['need_identification_number'],
+                (bool) $data['display_tax_label'],
+                [$this->defaultShopId]
+            ));
+            $this->getSharedStorage()->set($countryReference, new Country($countryId->getValue()));
```

Comment:
We don't store the object itself in the shared storage Because it's a big object to store for once, and mostly because by storing the ID only it forces us to re-fetch the entity each time we test it so it really ensure that the DB is up to date

Your current implementation works as long as you don't update the content, but if you modify the DB your stored object will not have been updated and will always believe the valid data are the one on creation